### PR TITLE
Debugg: state dumping for KProve

### PR DESCRIPTION
**Pull request by ehildenb**
This is a cleaned up version of the State Dumping that @mhhf implemented.

### CODE REVIEW

**Comment by ehildenb** [comment]
@daejunpark can you review the last commit which adds SMT axiomitization for exponentiation and let me know here if it needs any changes?

File: k-distribution/include/z3/exp.smt2 [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+(assert (forall ((x Int) (y Int)) (! (=> (>= y 1) (>= (expFunc x y) x      )) :pattern ((expFunc x y)))))
+(assert (forall ((x Int) (y Int)) (! (=> (>= y 2) (>= (expFunc x y) (* x x))) :pattern ((expFunc x y)))))
+
+;
+; smt_rpow
+;
+
+(declare-fun smt_rpow (Int Int Int Int) Int)
+(assert (forall ((z Int) (x Int) (y Int) (b Int)) (! (=> (= y 1) (= (smt_rpow z x y b) (div (+ (* z x) (div b 2)) b))) :pattern ((smt_rpow z x y b)))))
```
Comment (daejunpark) [inline k-distribution/include/z3/exp.smt2]: I do not remember what this was for. It seems not consistent with others. You can test if this is still needed.

**Comment by daejunpark** [comment]
@ehildenb I remember @mhhf and I wrote it together, except the line that I commented above.

BTW, another thought. This SMT prelude may be too specific to MakerDao verification. But it could be useful for the future reference when we have a similar issue for other verification projects. Not sure what is be the best place to put this.

**Comment by ehildenb** [comment]
Closing this for a sequence of smaller PRs.

**Pull request closed without merging**